### PR TITLE
Give callbacks its own go.mod

### PR DESCRIPTION
**Pull request by chatton**
## Description

Steps taken:
- added go.mod with local pin to ibc-go
- replace all occurences of github.com/cosmos/ibc-go/v7/... => github.com/cosmos/ibc-go/modules/...

closes: #4327

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: .github/workflows/callbacks.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Callbacks Module
+# This workflow runs when a PR is opened that targets code that is part of the callbacks module.
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/capability.yml'
```
Comment (damiannolan) [inline .github/workflows/callbacks.yml:6]:
```suggestion
      - '.github/workflows/callbacks.yml'
```

**Review by damiannolan — COMMENTED** [review]
Looks perfect to me! Will wait for tests to run and pass before approving :)

File: modules/apps/callbacks/ibc_middleware.go [hunk 2]
```diff
@@ -6,7 +6,7 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	capabilitytypes "github.com/cosmos/ibc-go/modules/capability/types"
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply modules/apps/callbacks/ibc_middleware.go:8]: Thanks, @damiannolan. I opened https://github.com/cosmos/ibc-go/issues/4342 to keep track of these tasks we need to do before the final release compatible with v7.3.0.

[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/callbacks/ibc_middleware.go:8]: Nice! Thanks @crodriguezvega

**Pull request merged**